be/src/runtime/types.cpp

```diff
@@ -445,4 +445,20 @@ TypeDescriptor TypeDescriptor::promote_types(const TypeDescriptor& type1, const
     return TypeDescriptor::create_varchar_type(TypeDescriptor::MAX_VARCHAR_LENGTH);
 }
 
+TypeDescriptor TypeDescriptor::promote_decimal_type(int precision, int scale) {
+    // decimal v3 only
+    if (precision <= 0 || precision > decimal_precision_limit<int128_t>) {
+        // if precision is invalid, use varchar
+        LOG(WARNING) << "failed to promote decimal type, use varchar. precision: " << precision;
```

Comment:
Can this log not be printed?

Reply:
this is abnormal, the log can tell us what the precision is after type merge.